sdk/core/core-https/src/nodeHttpsClient.ts

```diff
@@ -127,7 +129,7 @@ export class NodeHttpsClient implements HttpsClient {
             request
           };
 
-          let responseStream = shouldDecompress ? getDecodedResponseStream(res, headers) : res;
+          responseStream = shouldDecompress ? getDecodedResponseStream(res, headers) : res;
```

Comment:
@xirzec This causes change in behavior. The `responseStream` used in the catch block was never assigned to anything. Am assuming this is what you had originally wanted

Reply:
Looks like I didn't catch this problem when I reviewed #12038  -- nice find!